lib/db/schemaupdater.go

```diff
@@ -482,21 +486,20 @@ func (db *schemaUpdater) updateSchemato9(prev int) error {
 				if err := t.Put(sk, it.Key()); err != nil {
 					return err
 				}
-				if err := t.putFile(it.Key(), fi); err != nil {
-					return err
-				}
-				continue
+				changed = true
 			}
 		}
-		if prev == 8 {
-			// The transition to 8 already did the changes below.
-			continue
-		}
-		if fi.Blocks == nil {
-			continue
+		// 1.4.0 RCs had a bug where we'd reset the blocks, but
+		// not the blocks-hash (version 10).
+		if fi.Blocks == nil && fi.BlocksHash != nil {
```

Comment:
Isn't this every file if prev >= 8?

Reply:
In theory I'd still like to not just swallow incorrect blocks hashes, but in practice I agree: https://github.com/syncthing/syncthing/pull/6355

And as far as I understand the transition to 8 didn't change `BlocksHash` if `Blocks == nil` and before that `BlocksHash` didn't exist, so only files which were ignored/otherwise invalidated or deleted after upgrading to a 1.4.0 release candidate should be affected. The amount of panic events on sentry seems to suggest otherwise though.

Reply:
You’re raw unmarshalling the file info, so every file with blocks will have a block hash and nil block list